src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +530,44 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def checkGoogleGroupExists(groupEmail: String): Boolean = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val checkGroupRequest = directoryService.groups.get(groupEmail)
+    Try(executeGoogleRequest(checkGroupRequest)) match {
+      case Failure(_) => false
+      case Success(_) => true
+    }
+  }
+
+  override def createGoogleGroup(groupEmail: String, targetUserEmail: String): Option[String] = {
+    val newGroup = new Group().setEmail(groupEmail)
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+
+    val insertRequest = directoryService.groups.insert(newGroup)
+    val setEmail = Try(executeGoogleRequest[Group](insertRequest)) match {
+      case Failure(_) => {
+        logger.warn(s"Could not create new group called $groupEmail")
+        "" // return empty string
+      }
+      case Success(newGroupInfo) => {
+        // add targetUserEmail as member of google group - modeled after `override def addMemberToGroup` in workbench-libs HttpGoogleDirectoryDAO.scala
+        val member = new Member().setEmail(targetUserEmail).setRole(anonymizedGroupRole).setDeliverySettings(anonymizedGroupDeliverySettings)
```

Comment:
this is called `createGoogleGroup` but it does quite a bit more than that. for a DAO, i would expect separate generic and lightweight methods: one to create a google group and one to add a member to a group. there's also some logic specific to your use-case that i would be weary of including in something so generically named. could you pass in the anonymized group settings from wherever is calling this?

Reply:
that's a good point. do you think it makes more sense to rename this function to something more specific, like `createAnonymizedGoogleGroup`, or to break it down as you suggest into the separate functions that would be reusable for other Google group purposes in future?

Reply:
Ideally, it would be broken down into reusable code that doesn't contain any use-case specific logic. This DAO doesn't do a particularly good job at that, but the ones in workbench-libs do.

Because I don't think this code will be re-used in the near future, I would say it's okay to be in the middle-ground here, and keep the use-case specific logic but to break down the functions into more appropriate parts (which will help with readability IMO). i.e. you'll have:

* createAnonymizedGoogleGroup (like you suggested)
* addMemberToAnonymizedGoogleGroup

If you want, you can take it all the way and extract the anonymized portions too, but I wouldn't withhold my thumb for that.

Reply:
I broke out an addMemberToAnonymizedGroup function, and that left createGoogleGroup as a non-case-specific function, so I didn't change its name. it wouldn't be terribly hard to add the role settings as inputs to the addMemberToAnonymizedGroup function, I just wasn't sure of the best way to do so (scala-wise... in python I might want those settings to be kwargs inputs). would love if you'd take another look when you get a chance and see if this mostly did what you envisioned, @MatthewBemis